### PR TITLE
Allow more characters in filenames

### PR DESCRIPTION
**Pull request by rullzer**
Fix for #22247 

Basically a path with " in it was blocked on the receiving server when verifing the path.
Now since we no longer support ownCloud on windows we can update the list of allowed characters.

CC: @nickvergessen @icewind1991 @LukasReschke @PVince81

### CODE REVIEW

File: lib/public/constants.php [hunk 1]
```diff
@@ -70,5 +70,5 @@ class Constants {
 	/**
 	 * @since 8.0.0
 	 */
-	const FILENAME_INVALID_CHARS = "\\/<>:\"|?*\n";
+	const FILENAME_INVALID_CHARS = "\\/";
```
Comment (nickvergessen) [inline lib/public/constants.php]: Should document what changed when?

Comment (rullzer) [inline lib/public/constants.php]: yes you are right... fixed.

**Comment by MorrisJobke** [comment]
I think this also has impact to the clients - cc @DeepDiver1975 @dragotin

**Comment by rullzer** [comment]
@MorrisJobke yeah I tought so as well. But basically the federated share provider is the only one using this code. The other go trough the view->verifyPath

Basically on master you can right now already create files like `foo"bar` `foo>&<bar` etc.

So the clients already need to handle that. All this PR does is making sure we can also mount such federated shares.

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by PVince81** [comment]
Would like to get a second opinion from @DeepDiver1975 @icewind1991 about this change

**Comment by nickvergessen** [comment]
@PVince81 let me put it this way, the file is on your server, you should be able to share it.
Other people can already mount it from the link share, you can just not share it directly.
If we don't change the constant, then we should just stop using it for this case.
Any file that is in your instance should be sharable, the remote instance has to check if the file name is okay, not your local instance.

**Comment by rullzer** [comment]
@nickvergessen this check is actually on the remote instance. But that remote instance is also a 9.0 server which allows creation of such files. so it should also allow it now...

**Comment by nickvergessen** [comment]
This is basically a joke, because then you share `test` and have `test/te"st` and create the problem this way... anyway...

**Comment by icewind1991** [comment]
I think changing the constant is fine for now, we can look into it further for 9.1

:+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**